### PR TITLE
feat(v2/TableField): render Table field dropdown column variant

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR adds the dropdown column variant to Table fields.
Had to update dropdown menu's popper configuration to prevent the dropdown options from being clipped by the TableField container due to the presence of `overflowX`.

## Solution

**Features**:

- feat(SelectPopover): add fixed strategy, constrain width to ref elem
- feat: render dropdown column in Table fields

## Before & After Screenshots
Public form stories should be updated to display dropdown columns.

### CODE REVIEW

**Comment by karrui** [comment]
Merging merge conflicts before reopening for review

File: frontend/src/templates/Field/Table/TableField.tsx [hunk 1]
```diff
@@ -139,6 +142,13 @@ export const TableField = ({
           </Tbody>
         </Table>
       </Box>
+      {isMobile ? null : (
```
Comment (mantariksh) [inline frontend/src/templates/Field/Table/TableField.tsx:145]: do we not show the error in mobile?

File: frontend/src/templates/Field/Table/TableField.tsx [hunk 2]
```diff
@@ -139,6 +142,13 @@ export const TableField = ({
...
+        // Error will only render if table field has errors. Since the only possible errors are
+        // for required fields (for now), just render a hardcoded error message.
+        <FormErrorMessage my="0.75rem">
+          Please fill in the required fields.
```
Comment (mantariksh) [inline frontend/src/templates/Field/Table/TableField.tsx:149]: what might be a more sustainable solution here? perhaps `unique` the error messages and display all?

File: frontend/src/templates/Field/Table/TableField.tsx (hunk 1 above)
Comment (mantariksh) [reply frontend/src/templates/Field/Table/TableField.tsx:145]: saw the code in `ColumnCell` - would be good to add a comment here to explain!

**Review by mantariksh — APPROVED** [review]

File: frontend/src/components/Dropdown/components/SelectPopover/SelectPopover.tsx [hunk 3]
```diff
@@ -12,11 +12,57 @@ export const SelectPopoverProvider: FC = ({ children }): JSX.Element => {
...
+            state.placement.startsWith('right')
+              ? 'height'
+              : 'width'
+
+          if (!popperElement) return
+
+          const popperSize =
+            popperElement[
+              `offset${
+                widthOrHeight[0].toUpperCase() + widthOrHeight.slice(1)
```
[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/components/Dropdown/components/SelectPopover/SelectPopover.tsx:42]: Indeed. The perils of copy-pasting code on the INTERWEBS. Will update.

File: frontend/src/templates/Field/Table/TableField.tsx (hunk 2 above)
Comment (karrui) [reply frontend/src/templates/Field/Table/TableField.tsx:149]: I fear that would be too confusing. I will unique and show the first error I guess. Will eventually allow users to fix all the errors.

File: frontend/src/templates/Field/Table/TableField.tsx (hunk 1 above)
Comment (karrui) [reply frontend/src/templates/Field/Table/TableField.tsx]: 2da28ab

File: frontend/src/templates/Field/Table/TableField.tsx (hunk 2 above)
Comment (karrui) [reply frontend/src/templates/Field/Table/TableField.tsx]: [2da28ab](https://github.com/opengovsg/FormSG/pull/3577/commits/2da28ab18d00018a43e933e7138daf792b52a956)

File: frontend/src/components/Dropdown/components/SelectPopover/SelectPopover.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/components/Dropdown/components/SelectPopover/SelectPopover.tsx]: Didn't need this in the end, found a performance bug that caused too many rerenders when viewport width changed. Used[ ChakraUI's variant](https://github.com/chakra-ui/chakra-ui/blob/main/packages/popper/src/modifiers.ts#L9) instead. 

(Reason why we didn't use ChakraUI's own usePopper was due to their warning that the hook is undocumented and is subject to change...)
(also realize we ignored that when using FormControlProps LOL)

**Pull request merged**